lib/common/socketpool.c

```diff
@@ -271,6 +243,9 @@ void dispose_target(h2o_socketpool_target_t *target)
     free(target->url.host.base);
     free(target->url.path.base);
     free(target);
+    if (target->conf.data_for_balancer != NULL) {
```

Comment:
I assume that you do not need a `if` check here, since `free` does nothing if `NULL` is passed as an argument, and since this is not a performance-critical path.